### ISSUE TITLE
Wrong message ordering through proxy with encryption enabled

### ISSUE THREAD

**Issue by giacomoa** [kroxylicious/kroxylicious#838]
Consuming messages through proxy with encryption enabled results in random ordered messages and also seems to be some issues with timestamp in decrypted messages.

**To Reproduce**
1. Run the example described here: https://github.com/kroxylicious/kroxylicious/blob/main/kubernetes-examples/envelope-encryption/README.md until step 6:
```
echo "ibm: 99" | kaf -b minikube:30192 produce trades
```

2. Send some other message  through proxy
```
echo "ibm: 88" | kaf -b minikube:30192 produce trades
```
```
echo "ibm: 77" | kaf -b minikube:30192 produce trades
```
3. Consume messages through proxy:
```
# kaf -b minikube:30192 consume trades
Partition:   0
Offset:      0
Timestamp:   2023-12-20 17:17:35.829 +0100 CET
ibm: 99
Partition:   0
Offset:      1
Timestamp:   2023-12-20 17:18:34.859 +0100 CET
ibm: 77  # **wrong order, it's actually at offset 2**
Partition:   0
Offset:      2
Timestamp:   2023-12-20 17:18:29.44 +0100 CET
ibm: 88   # **wrong order, it's actually at offset 1**
```
4. Repeat step 2. and 3.

**Expected behavior**
Consumed messaged are ordered like produced with correct timestamp (i have only 1 partition in Kafka topic).

**Actual behavior**
Each consume action results in a different order (often wrong) and if I send two times the same message (with different timestamp) they appear with the same timestamp (and wrong order) when consuming through proxy

**Comment by k-wall** [comment]
Thanks for the feedback.   I'll be taking a look today.

**Comment by k-wall** [comment]
Yes, you are right, the consumption ordering is unstable.

```
kwall@Oslo kroxylicious % echo "ibm: 99" | kaf -b minikube:30192 produce trades
Sent record to partition 0 at offset 0.
kwall@Oslo kroxylicious % echo "ibm: 98" | kaf -b minikube:30192 produce trades
Sent record to partition 0 at offset 1.
kwall@Oslo kroxylicious % echo "ibm: 97" | kaf -b minikube:30192 produce trades
Sent record to partition 0 at offset 2.
kwall@Oslo kroxylicious % echo "ibm: 96" | kaf -b minikube:30192 produce trades
Sent record to partition 0 at offset 3.
kwall@Oslo kroxylicious % echo "ibm: 95" | kaf -b minikube:30192 produce trades
Sent record to partition 0 at offset 4.
kwall@Oslo kroxylicious %
kwall@Oslo kroxylicious %
kwall@Oslo kroxylicious % kaf -b minikube:30192 consume trades
Partition:   0
Offset:      0
Timestamp:   2023-12-21 10:42:54.135 +0000 GMT
ibm: 99
Partition:   0
Offset:      1
Timestamp:   2023-12-21 10:42:58.498 +0000 GMT
ibm: 98
Partition:   0
Offset:      2
Timestamp:   2023-12-21 10:43:02.552 +0000 GMT
ibm: 97
Partition:   0
Offset:      3
Timestamp:   2023-12-21 10:43:06.328 +0000 GMT
ibm: 96
Partition:   0
Offset:      4
Timestamp:   2023-12-21 10:43:09.863 +0000 GMT
ibm: 95
kwall@Oslo kroxylicious %
kwall@Oslo kroxylicious %
kwall@Oslo kroxylicious %
kwall@Oslo kroxylicious % kaf -b minikube:30192 consume trades
Partition:   0
Offset:      0
Timestamp:   2023-12-21 10:43:09.863 +0000 GMT
ibm: 95
Partition:   0
Offset:      1
Timestamp:   2023-12-21 10:42:54.135 +0000 GMT
ibm: 99
Partition:   0
Offset:      2
Timestamp:   2023-12-21 10:43:06.328 +0000 GMT
ibm: 96
Partition:   0
Offset:      3
Timestamp:   2023-12-21 10:42:58.498 +0000 GMT
ibm: 98
Partition:   0
Offset:      4
Timestamp:   2023-12-21 10:43:02.552 +0000 GMT
ibm: 97
```

**Comment by k-wall** [comment]
The issue is in io.kroxylicious.filter.encryption.inband.InBandKeyManager#decrypt. The order which receiver is called does not necessarily match the order of the records.

**Comment by k-wall** [comment]
With the fix on #839, I believe the issue is resolved.

```
% kaf -b localhost:9192 consume trades
Partition:   0
Offset:      0
Timestamp:   2023-12-21 11:18:54.514 +0000 GMT
ibm: 99
Partition:   0
Offset:      1
Timestamp:   2023-12-21 11:18:59.01 +0000 GMT
ibm: 98
Partition:   0
Offset:      2
Timestamp:   2023-12-21 11:19:02.444 +0000 GMT
ibm: 97
Partition:   0
Offset:      3
Timestamp:   2023-12-21 11:19:05.985 +0000 GMT
ibm: 96
Partition:   0
Offset:      4
Timestamp:   2023-12-21 11:19:09.609 +0000 GMT
ibm: 95
```
